engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+
+	ld hl, vChars2
+	ld de, IntroCharizard1GFX
+	lb bc, BANK(IntroCharizard1GFX), 8 tiles
+	call Request2bpp
+
+	ld hl, vFont
+	ld de, IntroCharizard2GFX tile $40
+	lb bc, BANK(IntroCharizard2GFX), 8 tiles
+	call Request2bpp
+
+	ld hl, vChars0
+	ld de, IntroCharizardFlamesGFX
+	lb bc, BANK(IntroCharizardFlamesGFX), 8 tiles
+	call Request2bpp
+
+	ld hl, wSpriteAnimDict
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_OMANYTE
+	ld [hli], a
+	ld a, SPRITE_ANIM_DICT_DEFAULT
+	ld [hli], a
+
+	ld a, 0
+	call DrawIntroCharizardGraphic
+
+	ld a, $80
+	ldh [hSCY], a
+
+	xor a
+	ldh [hSCX], a
+	ld [wGlobalAnimYOffset], a
+	ld [wGlobalAnimXOffset], a
+
+	xor a
+	ld [wIntroFrameCounter2], a
+	ld a, %00111111
+	ldh [rBGP], a
+	ld a, %11111111
+	ldh [rOBP0], a
+	ret
+
+IntroScene11:
+; scroll up to Charizard silhoutte, flash Blastoise and Venusaur
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	inc [hl]
+	and 1
+	ret z
+	call Intro_CheckSCYEvent
+	ld hl, hSCY
+	ld a, [hl]
+	and a
+	jr z, .next
+	inc [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	xor a
+	ld [wIntroFrameCounter1], a
+
+IntroScene12:
+; load Charizard palettes
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	srl a
+	srl a
+	and 3
+	ld e, a
+	ld d, 0
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	and a
+	jr z, .next
+	ldh [rBGP], a
+	ldh [rOBP0], a
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, $80
+	ld [wIntroFrameCounter1], a
+	ret
+
+.palettes:
+	db %01101010
+	db %10100101
+	db %11100100
+	db %00000000
+
+IntroScene13:
+; Charizard mouth open
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+
+	dec [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, 1
+	call DrawIntroCharizardGraphic
+	ld a, 4
+	ld [wIntroFrameCounter1], a
+	ret
+
+IntroScene14:
+; Charizard breathing fire
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+	dec [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, 2
+	call DrawIntroCharizardGraphic
+	ld a, 64
+	ld [wIntroFrameCounter1], a
+	xor a
+	ld [wIntroFrameCounter2], a
+; fallthrough
+
+IntroScene15:
+; Charizard mouth wide open / fireball starts
+	call Intro_AnimateFireball
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+	dec [hl]
+	ret
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	xor a
+	ld [wIntroFrameCounter1], a
+	ret
+
+IntroScene16:
+; continue fireball / fade out palettes
+	call Intro_AnimateFireball
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	swap a
+	and $07
+	ld e, a
+	ld d, $00
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	cp $ff
+	jr z, .next
+	ldh [rBGP], a
+	ldh [rOBP0], a
+	ret
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ret
+.palettes:
+	db %11100100
+	db %10010000
+	db %01000000
+	db %00000000
+	db -1
+
+Intro_BlankTilemapAndBGMap:
+	hlcoord 0, 0
+	ld bc, SCREEN_WIDTH * SCREEN_HEIGHT
+.blank_tilemap
+	ld [hl], 0
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .blank_tilemap
+
+	ld hl, wOverworldMapBlocks	; $c600, buffer
```

Comment:
If this and the line below aren't being used with the meaning of "overworld map blocks", add an alias label (even just `wc600`)